src/util/_codegen.py

```diff
@@ -20,6 +23,71 @@
 AUTOCAST_KEY = 'toloka_field_autocast'
 
 
+def _get_decorator_wrapper(wrapped_decorator):
+
+    @functools.wraps(wrapped_decorator)
+    def decorator_wrapper(func):
+        """Applies the decorator and wraps the resulting function-like object with the original type."""
+        decorated_func = wrapped_decorator(func)
+
+        if inspect.isasyncgenfunction(func):
+
+            @functools.wraps(decorated_func)
+            async def wrapped(*args, **kwargs):
+                async for item in decorated_func(*args, **kwargs):
+                    yield item
+
+        elif inspect.isgeneratorfunction(func):
+
+            @functools.wraps(decorated_func)
+            def wrapped(*args, **kwargs):
+                yield from decorated_func(*args, **kwargs)
+
+        elif inspect.iscoroutinefunction(func):
+
+            @functools.wraps(decorated_func)
+            async def wrapped(*args, **kwargs):
+                res = await decorated_func(*args, **kwargs)
+                while inspect.isawaitable(res):
```

Comment:
What is it for?

Reply:
Sorry, forgot about that: the true bug was in `expand`: fixed.